components/style/properties/longhand/inherited_box.mako.rs

```diff
@@ -50,6 +50,132 @@ ${helpers.single_keyword("image-rendering",
                          custom_consts=image_rendering_custom_consts,
                          animatable=False)}
 
+// Image Orientation
+// https://developer.mozilla.org/en-US/docs/Web/CSS/image-orientation
+<%helpers:longhand name="image-orientation"
+                   experimental="True"
+                   products="None"
+                   animatable="False">
+    use std::fmt;
+    use style_traits::ToCss;
+    use values::specified::Angle;
+
+    use values::NoViewportPercentage;
+    impl NoViewportPercentage for SpecifiedValue {}
+
+    #[derive(Clone, PartialEq, Copy, Debug)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct SpecifiedValue {
+        pub angle: Option<Angle>,
+        pub flipped: bool
+    }
+
+    impl ToCss for SpecifiedValue {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            if let Some(angle) = self.angle {
+                try!(angle.to_css(dest));
+                if self.flipped {
+                    dest.write_str(" flipped")
+                } else {
+                    Ok(())
+                }
+            } else {
+                if self.flipped {
+                    dest.write_str("flipped")
+                } else {
+                    dest.write_str("from-image")
+                }
+            }
+        }
+    }
+
+    pub mod computed_value {
+        use values::specified::Angle;
+
+        #[derive(Clone, PartialEq, Copy, Debug)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub enum T {
+            FromImage,
```

Comment:
Oops, sorry I didn't see the last commit about normalize_angle. Will look at it.